client/app/visualizations/table/renderer.less

```diff
@@ -0,0 +1,53 @@
+.table-visualization-container {
+  .ant-pagination.ant-table-pagination {
+    float: none;
+    display: block;
+    text-align: center;
+  }
+
+  .ant-table-body {
+    overflow-x: auto;
+  }
+
+  table {
+    .display-as-number,
+    .display-as-boolean,
+    .display-as-datetime,
+    .display-as-image {
+      width: 1%;
+      white-space: nowrap;
+    }
+
+    .table-visualization-spacer {
+      padding-left: 0;
+      padding-right: 0;
+    }
+
+    th {
+      .ant-input-search {
+        font-weight: normal;
+
+        .ant-input-suffix .anticon {
+          cursor: auto;
+        }
+      }
+
+      .sort-column-order-indicator {
+        @size: 12px;
+
+        display: inline-block;
+        vertical-align: middle;
+        min-width: @size;
+        height: @size;
+        font-size: @size * 3/4;
+        border-radius: @size / 2;
+        background: #c0c0c0;
+        text-align: center;
+        line-height: @size;
+        color: #fff;
+        padding: 0 @size * 1/4;
+        margin: 0 5px 0 0;
+      }
+    }
+  }
```

Comment:
A few differences in table style I noticed:

1. Cell height a pixel higher than original. This causes autoheight calculation to add a needless row ([example](https://redash-preview.netlify.com/dashboard/shared-dashboard)).
This should fix it:
```css
td {
  line-height: 1.4;
}
```
2. Ant table has a bottom border and no top border. I think the original looks better. Should change:
```less
td {
  border-bottom: 0;
  border-top: @the-variable;
}
```
3. The rules in `redash-table.less` should be revised. It's loaded but unused by table viz now.
One thing I'd like to apply is this, so the content aligns with the widget:
https://github.com/getredash/redash/blob/ed8b022e786b994e05a8087b9ae0db3a97e4da57/client/app/assets/less/redash/redash-table.less#L31-L37
4. Having the sort icons constantly visible makes for too much visual noise in widgets and dashboards.
This is my suggestion:
   1. Hidden by default `.ant-table-column-sorter .anticon.off { opacity: 0 }`
   2. Displayed on hover `thead:hover .off { opacity: 1 }`
   <br />
   <img src="https://user-images.githubusercontent.com/486954/61857597-27286c00-aecd-11e9-96f9-5e05354c5261.gif" width="400" />

   Bonus: Find a way to display the `.off` sibling of the `.on` as well. 

5. The table header cell [in this viz](https://deploy-preview-3963--redash-preview.netlify.com/queries/106#195) for some reason breaks line.

Reply:
1 and 2: should it be an app-wide fix or only for this visualization?
3, 4 - :+1: 
5: Previously headers had `white-space: no-wrap` CSS, but I noticed that it does not look good with long column names (for some column types like Number even 10-15 characters was too long). So I removed this CSS rule. But some columns (line Number, Boolean, etc.) have `width: 1%` - otherwise table will distribute available space accordingly to column's contents, and such columns will take too much space. With `width: 1%` they will be as narrow as possible, but wide enough to show the data. It works perfectly for numbers, booleans, date/time, etc. But It means that browser will wrap headers if needed to make that columns as narrow as possible.

Reply:
> 1 and 2: should it be an app-wide fix or only for this visualization?

Only for viz (tested the change app-wide and it's no good)

> I noticed that it does not look good with long column names (for some column types like Number even 10-15 characters was too long).

Can you show me an example of this?

Reply:
https://redash-preview.netlify.com/queries/187/source

I checked old CSS rules, and seems that previously idea was that long column titles should be truncated, but seems it didn't work.I can try to implements this: truncate long titles (show `...`), and add a tooltip with full text. WDYT?

Reply:
That would be perfect

Reply:
Actually, perhaps enable this as a viz setting?
Full text (nowrap) / shortened (ellipsis) / line break (normal).

Reply:
Hm, interesting idea :thinking: I'll do it if it will not take too much effort

Reply:
@ranbena I cannot align tooltip better, but it looks much better as before (at least, it shows ellipsis as intended). I think we can do another iteration on it later:
![image](https://user-images.githubusercontent.com/12139186/62142581-41ac7c00-b2f7-11e9-8c5a-9c7fe30b238c.png)
![image](https://user-images.githubusercontent.com/12139186/62142639-5a1c9680-b2f7-11e9-85ed-6d255dbb540e.png)

About cell height and auto-height calculation: almost always there will be some free space below the table. The only condition when there will be no extra space - if table height + height of widget header and footer will be a multiple of 50 (height of dashboard grid row). And there are no chance to impact on it: if we change line height, for some tables that space will decrease, but at the same time for other tables it will increase. Also, keep in mind that line height depends on font chosen by browser, and for different browsers it will be different. So for now I decided to keep everything as is to avoid adding extra code that will not improve the situation.

Reply:
About rules from `redash-table.less`: I restored some rules that were applied to table visualization, but this files cannot be removed ATM because it's used by some other components (visualization editors and also I saw it somewhere in `components` folder IIRC). Also, I decided to keep Ant's table styles (like cell paddings) - I think such things should be applied app-wide (e.g. on Queries/Dashboards list pages).

Reply:
> if we change line height, for some tables that space will decrease, but at the same time for other tables it will increase.

> Also, keep in mind that line height depends on font chosen by browser, and for different browsers it will be different.

All true. My reasoning is to just keep it the same as before, so autoheight widgets don't become larger and mess up ppl's dashboard alignments.